plugins/osdn/subnets.go

```diff
@@ -96,23 +104,25 @@ func (oc *OvsController) deleteNode(nodeName string) error {
 	return oc.Registry.DeleteSubnet(nodeName)
 }
 
-func (oc *OvsController) SubnetStartNode(mtu uint) (bool, error) {
-	err := oc.initSelfSubnet()
-	if err != nil {
-		return false, err
-	}
+func (oc *OvsController) SubnetStartNode(clusterNetworkCIDR string, mtu uint) (bool, error) {
+	var err error
 
 	// Assume we are working with IPv4
-	clusterNetworkCIDR, err := oc.Registry.GetClusterNetworkCIDR()
+	_, oc.clusterNet, err = net.ParseCIDR(clusterNetworkCIDR)
 	if err != nil {
-		log.Errorf("Failed to obtain ClusterNetwork: %v", err)
-		return false, err
+		return false, fmt.Errorf("Failed to parse cluster network address %s: %v", clusterNetworkCIDR, err)
 	}
+
 	servicesNetworkCIDR, err := oc.Registry.GetServicesNetworkCIDR()
 	if err != nil {
 		log.Errorf("Failed to obtain ServicesNetwork: %v", err)
 		return false, err
 	}
+
+	if err := oc.initSelfSubnet(oc.clusterNet); err != nil {
```

Comment:
Why does this get moved? In particular, this is the step that waits for the master to have finished initializing, so moving it to after the GetServicesNetworkCIDR() call might cause that call to fail.